lib/utils/parseSelector.mjs

```diff
@@ -4,12 +4,13 @@ import selectorParser from 'postcss-selector-parser';
  * @param {string} selector
  * @param {import('stylelint').PostcssResult} result
  * @param {import('postcss').Node} node
- * @param {(root: import('postcss-selector-parser').Root) => void} callback
- * @returns {string | undefined}
+ * @returns {import('postcss-selector-parser').Root | undefined}
  */
-export default function parseSelector(selector, result, node, callback) {
+export default function parseSelector(selector, result, node) {
```

Comment:
major

I think that changing the signature of that function will break many plugins that use the callback argument.
It's under utils so it's public.
Let's wait for v17?
Or fallback to the previous implementation if the callback is passed?
e.g.
```js
return callback ? selectorParser(callback).processSync(selector) : selectorParser().astSync(selector);
```

see https://github.com/search?q=%22stylelint%2Flib%2Futils%2FparseSelector%22&type=code

```suggestion
export default function parseSelector(selector, result, node, callback) {
```

Reply:
Good point. I've added a fallback via d81656b8e. 👍🏼